redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -187,110 +189,199 @@ inline void
+                                    "input power values";
+                getAverageMaximumValues(aResp, serviceName, averagePath,
+                                        maximumPath);
+            }
+            else
+            {
+                getServicePathValues(aResp, objectPath2, "", serviceName,
+                                     averagePath, maximumPath);
+            }
+        };
+
+    crow::connections::systemBus->async_method_call(
+        getServiceAndPathHandler, "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", objectPath,
+        interfaces);
+}
+
 /**
- * @brief Get power supply average, maximum and date values given chassis and
- * power supply IDs.
+ * @brief Get power supply average, maximum and date values given chassis
+ * and power supply IDs.
  *
  * @param[in] aResp - Shared pointer for asynchronous calls.
- * @param[in] chassisID - Chassis to which the values are associated.
- * @param[in] powerSupplyID - Power supply to which the values are associated.
+ * @param[in] inputHistoryItem - Array of object paths for input history.
  *
  * @return None.
  */
 inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
-                      const std::string& chassisID,
-                      const std::string& powerSupplyID)
+                      const std::vector<std::string>& inputHistoryItem)
 {
-    BMCWEB_LOG_DEBUG
-        << "Get power supply date/average/maximum input power values";
+    BMCWEB_LOG_DEBUG << "ENTER: getValues(...)";
+    for (const auto& item : inputHistoryItem)
+    {
+        BMCWEB_LOG_DEBUG << " inputHistoryItem: " << item;
+    }
+
     // Setup InputPowerHistoryItems values array.
     // It will have 0 to many date/timestamp, average, and maximum entries.
     aResp->res
         .jsonValue["Oem"]["IBM"]["InputPowerHistoryItems"]["@odata.type"] =
         "#OemPowerSupplyMetric.InputPowerHistoryItems";
 
-    const std::array<const char*, 2> interfaces = {
-        "org.open_power.Sensor.Aggregation.History.Average",
-        "org.open_power.Sensor.Aggregation.History.Maximum"};
+    std::string serviceName;
+    std::string averagePath;
+    std::string maximumPath;
 
-    crow::connections::systemBus->async_method_call(
-        [aResp, chassisID, powerSupplyID](
+    getServicePathValues(aResp, inputHistoryItem[0], inputHistoryItem[1],
+                         serviceName, averagePath, maximumPath);
+
+    BMCWEB_LOG_DEBUG << "EXIT: getValues(...)";
+}
+
+/**
+ * @brief Retrieves valid input history item.
+ *
+ * Not all power supplies support the power input history. Do not provide
+ * Redfish fields for input power history if no associated endpoint matches this
+ * chassis.
+ *
+ * @param asyncResp     Pointer to object holding response data
+ * @param powerSupplyPath Validated power supply path
+ * @param callback      Callback for next step to get valid chassis ID
```

Comment:
This description does not seem right.

Reply:
` * @param callback      Callback for next step to populate Redfish JSON.` 
?

Reply:
Sure that seems fine.  I just meant that the next step was not getting the chassis ID.